docs/spec/lightclient/attacks/evidence-handling.md

```diff
@@ -0,0 +1,250 @@
+`untrusted` at some height `h`. If the bisection algorithm succeeds, then the header `untrusted` is verified. 
+Headers that are downloaded as part of the bisection algorithm are stored in a store and they are also in 
+the verified state. Therefore, after the bisection algorithm successfully terminates we have a trace of 
+the light blocks ([] LightBlock) we obtained from the primary that we call primary trace.
+
+### Primary trace 
+
+The following invariant holds for the primary trace:
+
+- Given a `trusted` light block, target height `h`, and `primary_trace` ([] LightBlock): 
+    *primary_trace[0] == trusted* and *primary_trace[len(primary_trace)-1].Height == h* and 
+    successive light blocks are passing light client verification logic. 
+          
+
+### Witness with a conflicting header
+
+The verified header at height `h` is cross-checked with every witness as part of 
+[detection](https://github.com/informalsystems/tendermint-rs/tree/master/docs/spec/lightclient/detection). 
+If a witness returns the conflicting header at the height `h` the following procedure is executed to verify 
+if the conflicting header comes from the valid trace and if that's the case to create an attack evidence:
+
+
+#### Helper functions
+
+We assume the following helper functions:
+
+```go
+// Returns trace of verified light blocks starting from rootHeight and ending with targetHeight.
+trace(lightStore LightStore, rootHeight int64, targetHeight int64) LightBlock[]
+
+
+
+```
+
+
+```go
+func DetectLightClientAttacks(primary PeerID, 
+                              primary_trace []LightBlock, 
+                              witness PeerID) (LightClientAttackEvidence, LightClientAttackEvidence) {
+    primary_lca_evidence, witness_trace = DetectLightClientAttack(primary_trace, witness)
+    
+    witness_lca_evidence = nil
+    if witness_trace != nil {
+        witness_lca_evidence, _ = DetectLightClientAttack(witness_trace, primary)
+    }
+    return primary_lca_evidence, witness_lca_evidence
+}
+
+func DetectLightClientAttack(trace []LightBlock, peer PeerID) (LightClientAttackEvidence, []LightBlock) {
+
+    trusted = trace[0]
+    lightStore = new LightStore().Update(trusted, StateTrusted)
+
+    for i in 1..len(trace)-1 {
+        lightStore, result = VerifyToTarget(peer, lightStore , i) 
+   
+        if result == ResultFailure then return (nil, nil)
+        
+        current = lightStore.Get(i)
+        
+        // if obtained header is the same as in the trace we continue with a next height
+        if current.Header == trace[i].Header continue
+
+        // we have identified a conflicting header
+        attackType = nil
+        trustedBlock = trace[i-1]
+        conflictingBlock = trace[i]
+        
+        if !isPossibleBlock(trustedBlock, conflictingBlock) { 
+            attackType = LunaticAttack
+        } else if trustedBlock.Header.Round == conflictingBlock.Header.Round {
```

Comment:
This should be `current` instead of `trustedBlock` because `current` is at the same height as the `conflictingBlock` which is required for this verification logic.

Perhaps better terminology would be to call `trustedBlock` `commonBlock` because it is the last block that the trace and the peer have in common